sound/soc/sof/ipc4-topology.c

```diff
@@ -226,18 +233,18 @@ static int sof_ipc4_get_audio_fmt(struct snd_soc_component *scomp,
 			base_config[i].ibs, base_config[i].obs,
 			base_config[i].cpc, base_config[i].is_pages);
 
-	ret = sof_update_ipc_object(scomp, &base_config->audio_fmt,
+	ret = sof_update_ipc_object(scomp, available_fmt->sink_audio_fmts,
 				    SOF_IN_AUDIO_FORMAT_TOKENS, swidget->tuples,
-				    swidget->num_tuples, sizeof(*base_config),
+				    swidget->num_tuples, sizeof(struct sof_ipc4_audio_format),
```

Comment:
I find the use of "sink" very misleading here, especially with token name "IN_AUDIO". Why not just keep this as "in_audio_fmts". 

If we'd stick to source/sink, and as "struct sof_ipc4_available_audio_format" is describing a single module's configuration, the directionality should really be from the module's point of your. And that would lead us to call this field "source_audio_fmts" as a module gets data from sources.